packages/terra-overlay/README.md

```diff
@@ -4,9 +4,23 @@
 [![NPM version](https://badgen.net/npm/v/terra-overlay)](https://www.npmjs.org/package/terra-overlay)
 [![Build Status](https://badgen.net/travis/cerner/terra-core)](https://travis-ci.com/cerner/terra-core)
 
-The Terra Overlay component is a component that displays an overlay relative to the container that triggered the overlay. This component blocks interactions and fades out all elements of the triggering container.
+## Deprecation Notice
 
-A Loading Overlay is a subcomponent that displays an overlay with a spinner icon and loading message.
+The terra-overlay npm package is deprecated as of 3.62.0. terra-overlay will be removed from the repo in a future version, but the package will remain on npm.
+
+### What deprecation means
+Any apps that currently use terra-overlay will continue to work as-is.
+However, the terra-overlay package will no longer continue to be supported / developed.
+
+## Why?
+Terra Overlay design is a bad pattern and hence it is being deprecated.
```

Comment:
What do you mean by this is a bad pattern? being devils advocate, without any background context, I don't really feel like this sells me on why it should deprecated.

Also what is different about `terra-application-loading-overlay` that make it a better pattern?

Reply:
Yeah, i was going to get to that once we got some other things cleaned up, but it's not a bad pattern it's just an old pattern and now we want people to pull from terra-application instead. There will have to be some more tweaking of the wording here.

Reply:
I'd say the biggest reason for this deprecation is to integrate the presentation of overlays with the overall application framework to ensure accessibility. Improving the ease of use of these components is a plus, too.

Reply:
Updated here https://github.com/cerner/terra-core/pull/3357/commits/a63aa91b91c0854bcfb22e4a435822771f932e7f, with what @tbiethman mentioned ☝️.